packages/sdk-client-v3/src/utils/maskAuthData.ts

```diff
@@ -1,7 +1,8 @@
 import { MiddlewareRequest } from '../types/types'
+import cloneDeep from 'lodash.clonedeep'
 
 export default function maskAuthData(request: MiddlewareRequest) {
-  const _request = Object.assign({}, request)
+  const _request = cloneDeep(request)
```

Comment:
Instead of adding a `library` just to deep clone, I think a simple `JSON.parse(JSON.stringify(request))` will work.

Reply:
We can try as much as we can to avoid using third party `dependencies`.

This I believe the below snippet should also be sufficient, since `JSON.parse(JSON.stringify())` have issue with `functions` and `undefined` values, even though we don't have functions in our request object it is always good to be on the safer side.

We can run the the tests case again it as well.


```ts
import { MiddlewareRequest } from '../types/types'

export function deepCopy(request: MiddlewareRequest) {
  let map: MiddlewareRequest | Array<any>, index: string | number

  /** these checks can be ignored, since the function is not exposed */
  if (!request) return request
  if (typeof request !== 'object') return request

  if ('[object Array]' === Object.prototype.toString.apply(request)) {
    map = []
    for (index = 0; index < request.length; index++) {
      map[index] = deepCopy(request[index])
    }

    return map
  }

  map = Object.create(null)
  for (index in request) {
    if (request.hasOwnProperty(index)) {
      map[index] = deepCopy(request[index])
    }
  }

  return map
}
```

Reply:
You are right, there‘s no need for 3rd party lib: https://github.com/commercetools/commercetools-sdk-typescript/pull/887/commits/320fa7a2facd8d9b9e9026171fb91cbb716f917a. I think we could just leave the function simple since the request structure is defined in TS and should work fine.